packages/contracts/src/yield/context/TripleRateContext.sol

```diff
@@ -18,31 +18,40 @@ abstract contract TripleRateContext is CalcInterestMetadata, ITripleRateContext
     error TripleRateContext_TenorPeriodRegressionNotAllowed(uint256 tenorPeriod, uint256 newTenorPeriod);
 
     /**
-     * @notice Emits when the current Tenor Period is set, with its associated Reduced Rate.
+     * @notice Emits when the current [TenorPeriodRate] is set.
      *
-     * @param tenorPeriod The updated current Tenor Period.
      * @param reducedRate The updated Reduced Rate for the Tenor Period.
+     * @param tenorPeriod The updated current Tenor Period.
      */
-    event CurrentTenorPeriodAndRateChanged(uint256 tenorPeriod, uint256 reducedRate);
+    event CurrentTenorPeriodRateChanged(uint256 reducedRate, uint256 tenorPeriod);
 
+    /// @notice The Tenor, or Maturity Period, of this context.
     uint256 public immutable TENOR;
 
-    uint256 internal _currentTenorPeriod;
-    uint256 internal _currentReducedRate;
-
-    uint256 internal _previousTenorPeriod;
-    uint256 internal _previousReducedRate;
+    TenorPeriodRate internal _current;
+    TenorPeriodRate internal _previous;
 
+    /**
+     * @notice Creates a [TripleRateContext] instance.
+     *
+     * @param fullRateInPercentageScaled_ The [uint256] 'full' rate, in scaled percentage form.
+     * @param reducedRateInPercentageScaled_ The [uint256] 'reduced' rate, in scaled percentage form.
+     * @param fromPeriod_ The [uint256] Period to apply the reduced rate to.
+     * @param frequency_ The [uint256] interest frequency.
+     * @param tenor_ The [uint256] product maturity period, or tenor.
+     * @param decimals The [uint256] number of decimals that is the applied scaling.
+     */
     constructor(
         uint256 fullRateInPercentageScaled_,
         uint256 reducedRateInPercentageScaled_,
+        uint256 fromPeriod_,
         uint256 frequency_,
         uint256 tenor_,
         uint256 decimals
```

Comment:
suggest to create a Struct holding the Params for this.  quite a long list of uint256s!

Reply:
Doing, but it means creating an instance of the `struct` in the `constructor` of the `LiquidContinuousMultiTokenVault`.